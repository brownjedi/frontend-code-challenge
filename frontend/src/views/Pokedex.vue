<template>
  <div :class="styles.container">
    <Filter />
    <main :class="styles.main">
      <nav :class="styles.nav">
        <div :class="styles['nav-item']">
          <router-link to="/" :class="styles['nav-link']">All</router-link>
        </div>
        <div :class="styles['nav-item']">
          <router-link to="/favorites" :class="styles['nav-link']">Favorites</router-link>
        </div>
      </nav>
      <h1 :key="showOnlyFavorites" :class="styles.title">
        {{ showOnlyFavorites ? 'Say hi to your favorite Pokémon!' : 'What Pokémon are you looking for?' }}
      </h1>
      <Controls />
      <PokemonList :class="styles.list" :listView="listView" :showOnlyFavorites="showOnlyFavorites" />
    </main>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent, useCssModule } from 'vue'
import PokemonList from '@/components/PokemonList.vue'
import Controls from '@/components/Controls.vue'
import Filter from '@/components/Filter.vue'
import { useStore } from '@/store'

export default defineComponent({
  props: {
    showOnlyFavorites: {
      type: Boolean,
      default: undefined,
    },
  },
  components: {
    PokemonList,
    Controls,
    Filter,
  },
  setup() {
    const styles = useCssModule()
    const store = useStore()

    const color = computed(() => store.state.pokedex.ui.color)
    const listView = computed(() => store.state.pokedex.ui.listView)

    return { styles, color, listView }
  },
})
</script>

<style lang="scss" module>
.container {
  position: relative;
  display: flex;
  flex-direction: row;
}

.main {
  padding: 2rem;
  overflow: auto;
  max-width: 150rem;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  width: 100%;
}

.title {
  line-height: 1.25;
  align-self: center;
  transition: color 0.5s ease-in-out;
}

.list {
}

.nav {
  align-self: flex-end;
  display: inline-block;

  .nav-item {
    display: inline-block;
    line-height: 2rem;
    margin-left: 1rem;
  }

  .nav-link {
    text-decoration: none;
    color: var(--main-text-color);
    color: v-bind(color);
    font-weight: 500;
    transition: border-bottom-color 0.5s ease-in-out, color 0.5s ease-in-out;

    &:global(.router-link-active) {
      margin-bottom: -2px;
      border-bottom: 2px solid var(--main-text-color);
      /* border-bottom-color: v-bind(color); */
    }
  }
}
</style>
