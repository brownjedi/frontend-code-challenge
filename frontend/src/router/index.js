import Vue from 'vue'
import VueRouter from 'vue-router'
import Details from '../views/Details.vue'
import Pokedex from '../views/Pokedex.vue'

Vue.use(VueRouter)

const routes = [
  {
    path: '/',
    name: 'Pokédex',
    component: Pokedex,
  },
  {
    path: '/:id',
    name: 'Pokémon Details',
    component: Details,
  },
]

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes,
})

export default router
