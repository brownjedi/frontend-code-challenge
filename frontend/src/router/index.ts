import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router'
import Pokemon from '../views/Pokemon.vue'
import Pokedex from '../views/Pokedex.vue'

const routes: Array<RouteRecordRaw> = [
  {
    path: '/',
    name: 'Pokédex',
    component: Pokedex,
  },
  {
    path: '/favorites',
    name: 'Favorite Pokémon',
    component: Pokedex,
    props: { showOnlyFavorites: true },
  },
  {
    path: '/:name',
    name: 'Pokémon Details',
    component: Pokemon,
  },
]

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes,
})

export default router
