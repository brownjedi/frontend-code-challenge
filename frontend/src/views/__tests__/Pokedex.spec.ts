import { shallowMount } from '@vue/test-utils'
import Pokedex from '@/views/Pokedex.vue'
import router from '@/router'

describe('views/Pokedex.vue', () => {
  it('has the expected html structure', () => {
    const wrapper = shallowMount(Pokedex, {
      global: {
        plugins: [router],
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
