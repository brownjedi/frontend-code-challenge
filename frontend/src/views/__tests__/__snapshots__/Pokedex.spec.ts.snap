// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/Pokedex.vue has the expected html structure 1`] = `
<div
  class="container"
>
  <filter-stub />
  <div
    class="main"
  >
    <nav
      class="nav"
    >
      <div
        class="nav-item"
      >
        <router-link-stub
          ariacurrentvalue="page"
          class="nav-link"
          custom="false"
          replace="false"
          to="/"
        />
      </div>
      <div
        class="nav-item"
      >
        <router-link-stub
          ariacurrentvalue="page"
          class="nav-link"
          custom="false"
          replace="false"
          to="/favorites"
        />
      </div>
    </nav>
    <h1
      class="title"
    >
      What Pok√©mon are you looking for?
    </h1>
    <controls-stub />
    <card-list-stub
      class=""
    />
  </div>
</div>
`;
